doc/docusaurus/docs/troubleshooting.md

```diff
@@ -45,6 +58,45 @@ If your expected trace messages are missing, check the following [plugin flags](
 - If the `remove-trace` flag (default off) is on, all trace messages will be removed.
 - If the `preserve-logging` flag (default off) is off, the compiler may remove some trace messages during optimization.
 
+### Unexpected Evaluation Failure
+
+It is usually [advisable](./using-plutus-tx/compiling-plutus-tx) to use the `Strict` extension when writing Plutus Tx, which improves performance.
+However, be cautious, as this can result in unexpected evaluation failures.
+Consider the following script:
+
+```haskell
+{-# LANGUAGE Strict #-}
+
+data MyRedeemer = A | B
+
+myScript :: MyRedeemer -> ScriptContext -> Bool
+myScript redeemer ctx = case redeemer of
+  A -> condition1
+  B -> condition2
+  where
+    condition1, condition2 :: Bool
+    condition1 = ...
+    condition2 = if ... then True else traceError "condition 2 not met"
+```
+
+The `Strict` extension makes all bindings strict, which means even if `redeemer` matches `A`, `condition2` will still be evaluated.
```

Comment:
I had this bite me way too many times... 😢  Good idea to include it, even if it's not necessarily Plinth-specific.